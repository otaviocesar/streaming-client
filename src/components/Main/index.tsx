import * as S from './styles'

const Main = ({
  title = 'Streaming',
  description = 'Integração de Mídias Sociais'
}) => (
  <S.Wrapper>
    <S.Logo src="/img/logo.svg" alt="Um logotipo com a letra S" />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.LogoConnect
      src="/img/connect-midia.svg"
      alt="Arte que mostra a conexão com várias mídias sociais"
    />
  </S.Wrapper>
)

export default Main
