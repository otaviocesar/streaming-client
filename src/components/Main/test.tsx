import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /Streaming/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#1b1b1b' })
  })
})
