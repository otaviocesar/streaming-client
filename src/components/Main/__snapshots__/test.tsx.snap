// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #1b1b1b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 10rem;
  margin-bottom: 2rem;
}

.c4 {
  width: 70rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

<main
  class="c0"
>
  <img
    alt="Um logotipo com a letra S"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    Streaming
  </h1>
  <h2
    class="c3"
  >
    Integração de Mídias Sociais
  </h2>
  <img
    alt="Arte que mostra a conexão com várias mídias sociais"
    class="c4"
    src="/img/connect-midia.svg"
  />
</main>
`
